ddtrace/contrib/trace_utils.py

```diff
@@ -330,9 +330,26 @@ def activate_distributed_headers(tracer, int_config=None, request_headers=None,
 
     if override or (int_config and distributed_tracing_enabled(int_config)):
         context = HTTPPropagator.extract(request_headers)
+
         # Only need to activate the new context if something was propagated
-        if context.trace_id:
-            tracer.context_provider.activate(context)
+        if not context.trace_id:
+            return None
+
+        # Do not reactivate a context with the same trace id
+        # DEV: An example could be nested web frameworks, when one layer already
+        #      parsed request headers and activated them
+        #
+        # Example::
+        #
+        #     app = Flask(__name__)  # Traced via Flask instrumentation
+        #     app = DDWSGIMiddleware(app)  # Extra layer on top for WSGI
+        current_context = tracer.current_trace_context()
+        if current_context and current_context.trace_id == context.trace_id:
```

Comment:
maybe worth logging a debug message when this happens since this is a pretty subtle implicit behaviour that would be hard to debug without knowledge of this functionality

Reply:
yeah, I wasn't sure here. or maybe I should put this behind a config flag too?